backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/database/repos/CardRepository.kt

```diff
@@ -75,36 +73,48 @@ object CardRepository {
+        from: Instant,
+        until: Instant,
+        regionId: Int?
+    ): List<CardStatisticsResultModel> {
         val numAlias = Coalesce(Cards.id.count(), intLiteral(0)).alias("numCards")
-        val allCards = (Regions leftJoin Cards leftJoin Projects)
+        val cardsCreated = (Regions leftJoin Cards leftJoin Projects)
             .slice(Regions.id, numAlias)
-            .select { Cards.issueDate.greaterEq(dateStart) and Cards.issueDate.lessEq(dateEnd) }
+            .select { Cards.issueDate.greaterEq(from) and Cards.issueDate.less(until) }
             .groupBy(Regions.id)
             .alias("AllCards")
         val activeCards = (Regions leftJoin Cards)
             .slice(Regions.id, numAlias)
-            .select { Cards.firstActivationDate.isNotNull() and Cards.issueDate.greaterEq(dateStart) and Cards.issueDate.lessEq(dateEnd) }
+            .select {
+                Cards.firstActivationDate.isNotNull() and
+                    Cards.issueDate.greaterEq(from) and
+                    Cards.issueDate.less(until)
+            }
             .groupBy(Regions.id)
             .alias("ActiveCards")
-        val query =
-            Regions
-                .join(allCards, JoinType.LEFT, Regions.id, allCards[Regions.id])
-                .join(activeCards, JoinType.LEFT, Regions.id, activeCards[Regions.id])
-                .join(Projects, JoinType.LEFT, Projects.id, Regions.projectId)
-                .slice(Regions.name, Regions.prefix, allCards[numAlias], activeCards[numAlias])
-                .select(Projects.project eq project)
-        if (regionId !== null) {
-            query.adjustWhere { Op.build { Regions.id eq regionId } }
-        } else {
-            query.adjustWhere { Op.build { Projects.project eq project } }
-        }
+        val query = Regions
+            .join(cardsCreated, JoinType.LEFT, Regions.id, cardsCreated[Regions.id])
+            .join(activeCards, JoinType.LEFT, Regions.id, activeCards[Regions.id])
+            .slice(Regions.name, Regions.prefix, cardsCreated[numAlias], activeCards[numAlias])
+            .select(
+                if (regionId == null) {
+                    Regions.projectId eq projectId
+                } else {
+                    Regions.id eq regionId
+                }
+            )
             .orderBy(Regions.name, SortOrder.DESC)
             .orderBy(Regions.prefix, SortOrder.DESC)
-        val result = query
-            .map { CardStatisticsResultModel(region = it[Regions.prefix] + ' ' + it[Regions.name], cardsCreated = (it[allCards[numAlias]] ?: 0).toInt(), cardsActivated = (it[activeCards[numAlias]] ?: 0).toInt()) }
+        return query
+            .map {
+                CardStatisticsResultModel(
+                    region = it[Regions.prefix] + ' ' + it[Regions.name],
+                    cardsCreated = (it.getOrNull(cardsCreated[numAlias]) ?: 0).toInt(),
+                    cardsActivated = (it.getOrNull(activeCards[numAlias]) ?: 0).toInt()
```

Comment:
@f1sh1918 We should use the `getOrNull` method here, as the subscript operator seems to return null although its function declaration says otherwise.